crates/anvil-polkadot/tests/it/utils.rs

```diff
@@ -0,0 +1,161 @@
+use alloy_primitives::hex;
+use anvil_core::eth::EthRequest;
+use anvil_polkadot::{
+    api_server::{self, ApiHandle},
+    cmd::NodeArgs,
+    config::{AnvilNodeConfig, SubstrateNodeConfig},
+    opts::Anvil,
+    spawn,
+    substrate_node::service::Service,
+};
+use anvil_rpc::response::ResponseResult;
+use eyre::{Result, WrapErr};
+use futures::channel::oneshot;
+use parity_scale_codec::Decode;
+use polkadot_sdk::{
+    sc_cli::CliConfiguration,
+    sc_client_api::{BlockBackend, HeaderBackend},
+    sp_core::{storage::StorageKey, twox_128, H256},
+};
+use serde_json::{json, Value};
+
+pub struct TestNode {
+    pub service: Service,
+    pub api: ApiHandle,
+    pub _runtime_handle: tokio::runtime::Handle,
+}
+
+impl TestNode {
+    pub async fn new_custom<F>(anvil_args: Anvil, config_modifier: F) -> Result<Self>
+    where
+        F: FnOnce(&mut AnvilNodeConfig, &mut SubstrateNodeConfig),
+    {
+        let handle = tokio::runtime::Handle::current();
+
+        let (mut anvil_config, mut substrate_config) =
+            anvil_args.node.clone().into_node_config()?;
+
+        config_modifier(&mut anvil_config, &mut substrate_config);
+        let config = substrate_config.create_configuration(&anvil_args, handle.clone())?;
+        let (service, api) = spawn(anvil_config, config).await?;
+
+        Ok(Self { service, api, _runtime_handle: handle.clone() })
+    }
+
+    pub async fn new_default() -> Result<Self> {
+        let handle = tokio::runtime::Handle::current();
+
+        let mut anvil_args = Anvil {
+            global: foundry_cli::opts::GlobalArgs::default(),
+            node: NodeArgs::default(),
+            cmd: None,
+        };
+
+        anvil_args.node.no_mining = true;
+        anvil_args.node.mixed_mining = false;
+        anvil_args.node.port = 0; // auto-assign
+
+        let (mut anvil_config, mut substrate_config) =
+            anvil_args.node.clone().into_node_config()?;
+        anvil_config = anvil_config.set_silent(true);
+
+        let temp_dir = tempfile::tempdir()?;
```

Comment:
isn't this dropped at the end of the function?
we should keep it on the TestNode